test/expected/alter.out

```diff
@@ -194,3 +194,172 @@ SELECT * FROM _timescaledb_catalog.chunk WHERE id = 2;
 ERROR:  cannot rename column "time" of hypertable chunk "new_chunk_name"
 \set ON_ERROR_STOP 1
+-- Test that we can set tablespace of a hypertable
+\c single :ROLE_SUPERUSER
+SET client_min_messages = ERROR;
+DROP TABLESPACE IF EXISTS tablespace1;
+DROP TABLESPACE IF EXISTS tablespace2;
+SET client_min_messages = NOTICE;
+--test hypertable with tables space
+CREATE TABLESPACE tablespace1 OWNER :ROLE_DEFAULT_PERM_USER LOCATION :TEST_TABLESPACE1_PATH;
+CREATE TABLESPACE tablespace2 OWNER :ROLE_DEFAULT_PERM_USER LOCATION :TEST_TABLESPACE2_PATH;
+\c single :ROLE_DEFAULT_PERM_USER
+-- Test that we cannot directly change chunk tablespace
+\set ON_ERROR_STOP 0
+ALTER TABLE public.new_chunk_name SET TABLESPACE tablespace1;
+ERROR:  operation not supported on chunk tables
+\set ON_ERROR_STOP 1
+-- drop all tables to make checking the tests below easier
+DROP TABLE alter_before;
+DROP TABLE alter_after;
+-- should return 0 rows
+SELECT tablename, tablespace FROM pg_tables
+WHERE tablename = 'hyper_in_space' OR tablename LIKE '\_hyper\__\__\_chunk' ORDER BY tablename;
+ tablename | tablespace 
+-----------+------------
+(0 rows)
+
+CREATE TABLE hyper_in_space(time bigint, temp float, device int);
+SELECT create_hypertable('hyper_in_space', 'time', 'device', 4, chunk_time_interval=>1);
+NOTICE:  adding not-null constraint to column "time"
+ create_hypertable 
+-------------------
+ 
+(1 row)
+
+INSERT INTO hyper_in_space(time, temp, device) VALUES (1, 20, 1);
+INSERT INTO hyper_in_space(time, temp, device) VALUES (3, 21, 2);
+INSERT INTO hyper_in_space(time, temp, device) VALUES (5, 23, 1);
+SELECT tablename FROM pg_tables WHERE tablespace = 'tablespace1' ORDER BY tablename;
+ tablename 
+-----------
+(0 rows)
+
+SET default_tablespace = tablespace1;
+-- should be inserted in tablespace1 which is now default
+INSERT INTO hyper_in_space(time, temp, device) VALUES (11, 24, 3);
+SELECT tablename, tablespace FROM pg_tables
+WHERE tablename = 'hyper_in_space' OR tablename LIKE '\_hyper\__\__\_chunk' ORDER BY tablename;
+    tablename     | tablespace  
+------------------+-------------
+ _hyper_3_5_chunk | 
+ _hyper_3_6_chunk | 
+ _hyper_3_7_chunk | 
+ _hyper_3_8_chunk | tablespace1
+ hyper_in_space   | 
+(5 rows)
+
+ALTER TABLE hyper_in_space SET TABLESPACE tablespace1;
+SELECT tablename FROM pg_tables WHERE tablespace = 'tablespace1' ORDER BY tablename;
+    tablename     
+------------------
+ _hyper_3_5_chunk
+ _hyper_3_6_chunk
+ _hyper_3_7_chunk
+ _hyper_3_8_chunk
+ hyper_in_space
+(5 rows)
+
+-- should be inserted in an existing chunk in the new tablespace,
+-- no new chunks
+INSERT INTO hyper_in_space(time, temp, device) VALUES (5, 27, 1);
+-- the new chunk should be create in the new tablespace
+INSERT INTO hyper_in_space(time, temp, device) VALUES (8, 24, 2);
+SELECT tablename, tablespace FROM pg_tables
+WHERE tablename = 'hyper_in_space' OR tablename LIKE '\_hyper\__\__\_chunk' ORDER BY tablename;
```

Comment:
should I keep this method of checking what is a hypertable chunk or use 
`tablename ~ '_hyper_\d+_\d+_chunk'` instead?

Reply:
what you have is fine